src/control/RotateControl.ts

```diff
@@ -0,0 +1,313 @@
+    this._yMotion = new Motion({ duration, range: DEFAULT.PITCH_RANGE, easing });
+  }
+
+  /**
+   * Destroy the instance and remove all event listeners attached
+   * @returns {void}
+   */
+  public destroy(): void {
+    this.disable();
+  }
+
+  /**
+   * Update control by given deltaTime
+   * @param deltaTime Number of milisec to update
+   * @returns {void}
+   */
+  public update(deltaTime: number): void {
+    const camera = this._view3D.camera;
+    const xMotion = this._xMotion;
+    const yMotion = this._yMotion;
+
+    const delta = new THREE.Vector2(
+      xMotion.update(deltaTime),
+      yMotion.update(deltaTime),
+    );
+
+    camera.yaw += delta.x;
+    camera.pitch += delta.y;
+  }
+
+  /**
+   * Resize control to match target size
+   * @param {object} size New size to apply
+   * @param {number} [size.width] New width
+   * @param {number} [size.height] New height
+   */
+  public resize(size: { width: number; height: number }) {
+    this._screenScale.set(360 / size.width, 180 / size.height);
+  }
+
+  /**
+   * Enable this input and add event listeners
+   * @returns {void}
+   */
+  public enable(): void {
+    if (this._enabled) return;
+
+    const targetEl = this._view3D.renderer.canvas;
+
+    targetEl.addEventListener(BROWSER.EVENTS.MOUSE_DOWN, this._onMouseDown);
+
+    targetEl.addEventListener(BROWSER.EVENTS.TOUCH_START, this._onTouchStart, { passive: true });
+    targetEl.addEventListener(BROWSER.EVENTS.TOUCH_MOVE, this._onTouchMove, { passive: this._view3D.scrollable });
+    targetEl.addEventListener(BROWSER.EVENTS.TOUCH_END, this._onTouchEnd);
+
+    this._enabled = true;
+    this.sync();
+
+    this.trigger(CONTROL_EVENTS.ENABLE);
+  }
+
+  /**
+   * Disable this input and remove all event handlers
+   * @returns {void}
+   */
+  public disable(): void {
+    if (!this._enabled) return;
+
+    const targetEl = this._view3D.renderer.canvas;
+
+    targetEl.removeEventListener(BROWSER.EVENTS.MOUSE_DOWN, this._onMouseDown);
+    window.removeEventListener(BROWSER.EVENTS.MOUSE_MOVE, this._onMouseMove);
+    window.removeEventListener(BROWSER.EVENTS.MOUSE_UP, this._onMouseUp);
+
+    targetEl.removeEventListener(BROWSER.EVENTS.TOUCH_START, this._onTouchStart);
+    targetEl.removeEventListener(BROWSER.EVENTS.TOUCH_MOVE, this._onTouchMove);
+    targetEl.removeEventListener(BROWSER.EVENTS.TOUCH_END, this._onTouchEnd);
+
+    this._enabled = false;
+
+    this.trigger(CONTROL_EVENTS.DISABLE);
+  }
+
+  /**
+   * Synchronize this control's state to given camera position
+   * @returns {void}
+   */
+  public sync(): void {
+    const camera = this._view3D.camera;
+
+    this._xMotion.reset(camera.yaw);
+    this._yMotion.reset(camera.pitch);
+  }
+
+  private _onMouseDown = (evt: MouseEvent) => {
+    if (evt.button !== BROWSER.MOUSE_BUTTON.LEFT) return;
+
+    const targetEl = this._view3D.renderer.canvas;
+    evt.preventDefault();
+
+    if (!!targetEl.focus) {
```

Comment:
If it is in an iframe, focus may not be available. with `evt.preventDefault();`

Reply:
I've just tested it on different devices and looks like it works fine.
I remember I added that to prevent some kind of bug that I can't remember.
Anyway, I'll keep that in mind.